_posts/2018/2018-08-05-delegate-to-simplify-your-code.md

```diff
@@ -0,0 +1,90 @@
+---
+title: Delegate to simplify your code
+description: 'There’s both a Ruby and a Rails version'
+layout: article
+category: ruby
+image:
+  base: '2018/delegate-to-simplify-your-code'
+  alt: 'Graffiti arrows'
+  source: 'https://unsplash.com/photos/Ze3l02aRwmE'
+  credit: 'Annie Spratt'
+---
+
+One way of thinking about object-oriented programming is as passing messages between objects.
+
+In many cases you may want to surface a public method on a associated object as if it was a method on the original object.
+
+The two main ways to do this are:
+
+ * the `Forwardable` functionality in the the standard Ruby library, [documentation here](http://ruby-doc.org/stdlib-2.5.1/libdoc/forwardable/rdoc/Forwardable.html#method-i-def_delegator).
+ * `delegate`, an Active Support core extension (availible if you’re using Rails) [documentation here](https://github.com/rails/rails/blob/master/activesupport/lib/active_support/core_ext/module/delegation.rb).
+
+## Instead of…
+
+... creating new methods to call directly to associated objects.
+
+```ruby
+# Inside Rails
+class Workspace < ApplicationRecord
+  belongs_to :user
+
+  def user_email
+    user.email
+  end
+end
+
+# Plain Ruby
+class Workspace
+  attr_reader :user
+
+  def initialize(user)
+    @user = user
+  end
+
+  def user_email
+    @user.email
+  end
+end
+```
+
+
+## Use…
+
+...the `Forwardable` functionality in the the standard Ruby library. Or `delegate`, which is one of ActiveSupport’s core extensions.
```

Comment:
Since you've explained what they are already above, I'd say:

*...the the `Forwardable` functionality or the `delegate` method.